HTML5: Enable mbedTLS module for Crypto object, force disable Asset Library

akien-mga opened the pull request:
Increases the size of the wasm by around 3% (~300-350 KiB).

This enables using the Crypto object for hashing, signing and encryption,
and therefore reduces the gap between the features of the HTML5 platform
and other platforms.

Closes https://github.com/godotengine/godot-proposals/issues/3574.

akien-mga commented:
Also removed the matching notes from the documentation. Didn't test that the features actually work fine yet though.

There's also still a similar note in WebSocketServer but I think this one is due to browser limitations and not mbedTLS? @Faless

akien-mga commented:
The Asset Library might need more work to be usable in the Web build, with this PR it's now enabled but spams warnings:
```
WARNING: HTTPClient polled multiple times in one frame, but request cannot progress more than once per frame on the HTML5 platform. [godot.tools.js:369:18](http://localhost:3000/godot.tools.js)
   at: poll (platform/javascript/http_client_javascript.cpp:270) - HTTPClient polled multiple times in one frame, but request cannot progress more than once per frame on the HTML5 platform. [godot.tools.js:369:18](http://localhost:3000/godot.tools.js)
```
And it doesn't show any asset.

Faless commented:
As mentioned in chat, the asset library cannot work right now because browsers requires [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) which github (where the assets are hosted) do not allow.

To support the asset library in the browsers we would need to either host or mirror the assets somewhere else, allowing CORS (at least from our domain).

akien-mga committed:
HTML5: Always disable Asset Library

GitHub doesn't allow CORS so we can't download assets from it.

There'd also be more work needed for the Asset Library plugin to be
usable in the Web editor even if that was supported.

akien-mga committed:
HTML5: Enable mbedTLS module for Crypto object

Increases the size of the wasm by around 3% (~300-350 KiB).

This enables using the Crypto object for hashing, signing and encryption,
and therefore reduces the gap between the features of the HTML5 platform
and other platforms.

Closes https://github.com/godotengine/godot-proposals/issues/3574.

akien-mga commented:
I added a comment to always disable the Asset Library on HTML5 without relying on the availability of `StreamPeerSSL` only.

akien-mga commented:
Cherry-picked for 3.5.
